Task/top 100 imdb movies

mratunjya opened the pull request.

mratunjya committed:
Movie Language Model Created

mratunjya committed:
Movie Rating Model Created

mratunjya committed:
Some updations in models

mratunjya committed:
Rake Task to save top 100 movies data using omdb api

ankitkhadria commented on lib/tasks/omdb.rake:
```diff
@@ -0,0 +1,250 @@
+  task save_top_100_movies_to_db: [:environment, :omdb_api_setup] do
+    # Task to fetch details of the top 100 movies from OMDB API and store them in the database
+
+    top_100_movies_imdb_id = [
```
rename variable name to `top_movies_imdb_ids`

ankitkhadria commented on lib/tasks/omdb.rake:
```diff
@@ -0,0 +1,250 @@
+        movie.year = year
+
+        # Calling methods to save related details to the database
+        save_actors(actors, imdb_id)
```
Try to use `accepts_nested_attributes_for`

https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html

ankitkhadria replied:
same for other associations

ankitkhadria commented on lib/tasks/omdb.rake:
```diff
@@ -0,0 +1,250 @@
+    api_key = "1f724ce8"
+    base_url = "http://www.omdbapi.com/"
+
+    ENV['api_url'] = "#{base_url}?apikey=#{api_key}"
```
move to the respective env file, right now you are using it in your local system then you can use `config/environments/development.rb`

ankitkhadria commented on lib/tasks/omdb.rake:
```diff
@@ -0,0 +1,250 @@
+
+    url = "#{api_url}&#{url_params}"
+    uri = URI.parse(url)
+    response = Net::HTTP.get_response(uri)
```
Add a check here like if the API is returning 200 or not, then only you'll return response, otherwise you need to handle the retry logic or restart the task.

mratunjya committed:
Added Inverse of Model Association, accepts_nested_attributes_for added in movie model

When we declare inverse_of on an association, it tells Rails that the association on the other side also knows about this association, and it helps to avoid unnecessary database queries.

mratunjya committed:
Minor Improvements in variable naming, and error handling for db as well as for omdb api response

mratunjya committed:
Merge pull request #5 from mratunjya/pr-review-changes/task/top-100-imdb-movies

Pr review changes/task/top 100 imdb movies
